File: extract_pdfs.js
import fs from 'fs';
import path from 'path';
import { createRequire } from 'module';
const require = createRequire(import.meta.url);
const pdf = require('pdf-parse');

const pdfDir = '/Users/evelynmollmann/.gemini/antigravity/scratch/aih-consultation/Protocolos de Solicitacao de Exames';
const outDir = '/Users/evelynmollmann/.gemini/antigravity/scratch/aih-consultation/documentos_referencia_txt';

if (!fs.existsSync(outDir)) {
    fs.mkdirSync(outDir);
}

const files = fs.readdirSync(pdfDir).filter(f => f.endsWith('.pdf'));

async function processPdfs() {
    for (const file of files) {
        const dataBuffer = fs.readFileSync(path.join(pdfDir, file));
        try {
            const data = await pdf(dataBuffer);
            fs.writeFileSync(path.join(outDir, file.replace('.pdf', '.txt')), data.text);
            console.log(`Processed: ${file}`);
        } catch (e) {
            console.error(`Error processing ${file}:`, e.message);
        }
    }
}

processPdfs();

File: extract_pdfs.py
import sys, os
from pypdf import PdfReader

pdf_dir = "/Users/evelynmollmann/.gemini/antigravity/scratch/aih-consultation/Protocolos de Solicitacao de Exames"
out_dir = "/Users/evelynmollmann/.gemini/antigravity/scratch/aih-consultation/documentos_referencia_txt"

os.makedirs(out_dir, exist_ok=True)

for file in os.listdir(pdf_dir):
    if file.endswith(".pdf"):
        pdf_path = os.path.join(pdf_dir, file)
        out_path = os.path.join(out_dir, file.replace(".pdf", ".txt"))
        try:
            reader = PdfReader(pdf_path)
            text = ""
            for page in reader.pages:
                ext = page.extract_text()
                if ext:
                    text += ext + "\n"
            with open(out_path, "w", encoding="utf-8") as f:
                f.write(text)
            print("Processed:", file)
        except Exception as e:
            print("Error:", file, repr(e))
